security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/BaseBuilder.java

```diff
@@ -0,0 +1,494 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.security.providers.oidc.common;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Collections;
+
+import io.helidon.common.Builder;
+import io.helidon.common.Errors;
+import io.helidon.common.configurable.Resource;
+import io.helidon.config.Config;
+import io.helidon.config.metadata.ConfiguredOption;
+import io.helidon.security.jwt.jwk.JwkKeys;
+
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonReaderFactory;
+
+/**
+ * Base builder of the OIDC config components.
+ */
+abstract class BaseBuilder<B extends BaseBuilder<B, T>, T extends TenantConfig> implements Builder<B, T> {
+
+    static final String DEFAULT_SERVER_TYPE = "@default";
+    static final String DEFAULT_BASE_SCOPES = "openid";
+    static final String DEFAULT_REALM = "helidon";
+    static final boolean DEFAULT_JWT_VALIDATE_JWK = true;
+    static final int DEFAULT_TIMEOUT_SECONDS = 30;
+    private static final JsonReaderFactory JSON = Json.createReaderFactory(Collections.emptyMap());
+
+    @SuppressWarnings("unchecked")
```

Comment:
this is no longer needed. Return `identity()` instead (supported by Builder in common)